tools/delete_candidate_info.php

```diff
@@ -0,0 +1,445 @@
+
+    // Print from session
+    echo "Session\n";
+    $result = $DB->pselect('SELECT * FROM session WHERE ID=:id', array('id' => $sessionID));
+    print_r($result);
+
+    // Print from feedback
+    echo "Behavioural Feedback\n";
+    $result = $DB->pselect(
+        'SELECT * from feedback_bvl_thread WHERE SessionID =:sid',
+        array('sid' => $sessionID)
+    );
+    print_r($result);
+    $feedbackIDs = $DB->pselect(
+        'SELECT FeedbackID from feedback_bvl_thread WHERE SessionID =:sid',
+        array('sid' => $sessionID)
+    );
+    foreach ($feedbackIDs as $id) {
+        $result = $DB->pselect(
+            'SELECT * from feedback_bvl_entry WHERE FeedbackID=:fid',
+            array('fid' => $id['FeedbackID'])
+        );
+        print_r($result);
+    }
+
+    // IF CONFIRMED, DELETE TIMEPOINT
+    if ($confirm) {
+        // Delete each instrument instance
+        foreach ($instruments as $instrument) {
+            echo "Deleting $instrument.\n";
+            $DB->delete($instrument['Test_name'], array('CommentID' => $instrument['CommentID']));
+
+            // Delete from conflicts
+            $DB->delete('conflicts_unresolved', array('CommentId1' => $instrument['CommentID']));
+            $DB->delete('conflicts_unresolved', array('CommentId2' => $instrument['CommentID']));
+            $DB->delete('conflicts_resolved', array('CommentId1' => $instrument['CommentID']));
+            $DB->delete('conflicts_resolved', array('CommentId2' => $instrument['CommentID']));
+        }
+        // Delete from flag
+        echo "Deleting from flag.\n";
+        $DB->delete('flag', array('SessionID' => $sessionID));
+
+        // Delete from session
+        echo "Deleting from session.\n";
+        $DB->delete('session', array('ID' => $sessionID));
+
+        // Delete from feedback
+        echo "Deleting from feedback.\n";
+        $DB->delete('feedback_bvl_thread', array('SessionID' => $sessionID));
+        foreach ($feedbackIDs as $id) {
+            $DB->delete('feedback_bvl_entry', array('FeedbackID' => $id));
+        }
+    }
+}
+
+function resetInstrument($commentID, $confirm) {
+
+    $DB =& Database::singleton();
+
+    $testName = $DB->pselectOne(
+        'SELECT Test_name FROM flag WHERE CommentID=:cid',
+        array('cid' => $commentID)
+    );
+
+    $columns = $DB->pselect(
+        'SELECT `COLUMN_NAME` FROM `INFORMATION_SCHEMA`.`COLUMNS` WHERE `TABLE_NAME`=:test',
+        array('test' => $testName)
+    );
+
+    // Fields from the $testName table to NOT reset
+    $excluded = array('CommentID');
+
+    $set = array();
+    $where = array('CommentID' => $commentID);
+
+    echo "Fields in $testName to set to null: \n";
+    foreach ($columns as $column) {
+        if (!in_array($column['COLUMN_NAME'], $excluded)) {
+            $current = array($column['COLUMN_NAME'] => NULL);
+            $set = array_merge($set, $current);
+            echo $column['COLUMN_NAME'] . "\n";
+        }
+    }
```

Comment:
This will likely fail if a field is not nullable, for example Testdate and Data_entry_completion_status.